### ISSUE TITLE
✨ Better logging for cache mount conflicts

### ISSUE THREAD

**Issue by nipuna-perera** [dagger/dagger#5617]
### What are you trying to do?

I had a scenario while testing different builder images where I had to set different `Owner` for certain tool chain images.

```Go
...
WithMountedCache("/mydir/.m2", c.CacheVolume("m2"), dagger.ContainerWithMountedCacheOpts{
  Owner: "jenkins:jenkins",
}).
...
```

The current implementation of this silently ignores the new cache if a previous cache (with different owner) exists. https://pkg.go.dev/dagger.io/dagger#ContainerWithMountedCacheOpts

It would be helpful for debugging pipeline issues if this conflict was logged as a warning.

### Why is this important to you?

This is a common issue that pipeline developers will run into when testing images that require specific `Owner` and `Group`

### How are you currently working around this?

I had to `docker rm -fv dagger-engine` to start with a clean slate so the new cache mount is picked up properly.

More context: https://discord.com/channels/707636530424053791/1138601483751936041

**Comment by gerhard** [comment]
What are you thoughts on adding an owner value to `CacheVolume()`, e.g.
1. `c.CacheVolume("m2-jenkins")`
2. `c.CacheVolume("m2-owner2")`

**Comment by nipuna-perera** [comment]
> What are you thoughts on adding an owner value to `CacheVolume()`, e.g.
> 
> 1. `c.CacheVolume("m2-jenkins")`
> 2. `c.CacheVolume("m2-owner2")`

Yes the user can do that but the intention of this request is to help the user identify when the engine runs into the conflict scenario and ignores the cache.

**Comment by skycaptain** [comment]
I believe I encountered the same problem. Here's the situation: I have two Dagger modules. The first builds my software as a non-root user and uses cache volumes. However, this module only handles the software's build. The second module uploads the cache to my Artifactory server. When I attempted to mount the Cache Volume using the same key, I received an empty directory. Through debugging, I discovered that I needed to mount the Cache Volume with the same owner for the files to even appear. Therefore, the second module must know the original owner of the files. For me, this presents the following issues:

- A cache cannot be shared between modules unless both modules use a commonly known owner.
- A cache volume becomes unusable if the owner of the image changes. In my first module, I allow users to use their own base image, which might utilize different non-root users, e.g. in my own image I commonly use `1000:1000`, but I know of at least one other popular image, that’s using `30000:30000`.
- As previously mentioned, it would be helpful to confirm whether this limitation is a bug or if it is functioning as designed in the docs
